crates/leon/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+//! Dead-simple string templating.
+//!
+//! Leon parses a template string into a list of tokens, and then substitutes
+//! provided values in. Unlike other templating engines, it is extremely simple:
+//! it supports no logic, only replaces. It is even simpler than `format!()`,
+//! albeit with a similar syntax.
+//!
+//! # Syntax
+//!
+//! ```plain
+//! it is better to rule { group }
+//! one can live {adverb} without power
+//! ```
+//!
+//! A replacement is denoted by `{` and `}`. The contents of the braces, trimmed
+//! of any whitespace, are the key. Any text outside of braces is left as-is.
+//!
+//! To escape a brace, use `\{` or `\}`. To escape a backslash, use `\\`. Keys
```

Comment:
Can we have escapes in the keys?

E.g.

```plain
{ key\\\{\} }
```

In this case, will the key be `key\\\{\}` or `key\{}`?

Reply:
we can't. didn't much see the point

Reply:
> we can't. didn't much see the point

That sounds reasonable and it is actually what I am expecting.

Maybe we should clarify it in the doc that escape is for the literal/text only and is not available for keys?

Reply:
It does say so! "Keys cannot contain escapes"

Reply:
Sorry I missed that.